pyquil/paulis.py

```diff
@@ -289,11 +316,35 @@ def __init__(self, terms):
         """
         :param list terms: A list of PauliTerms.
         """
+        if not (isinstance(terms, list) and all([isinstance(term, PauliTerm) for term in terms])):
+            raise ValueError("PauliSum's are currently constructed from lists of PauliTerms.")
         if len(terms) == 0:
             self.terms = [0.0 * ID]
         else:
             self.terms = terms
 
+    def __eq__(self, other):
+        """Equality testing to see if two PauliSum's are equivalent.
+
+        :param PauliSum other: The PauliSum to compare this PauliSum with.
+        :return: True if other is equivalent to this PauliSum, False otherwise.
+        :rtype: bool
+        """
+        if not isinstance(other, PauliSum):
+            return False
+        elif len(self.terms) != len(other.terms):
```

Comment:
You can return `False` but raise a warning that the list are not of equal length. Nice usability feature.

Reply:
@jotterbach What do you mean `return False` but `raise` a warning? Aren't those two things mutually exclusive?